flow/hooks/log_operations.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Builtin execution hook for basic logging."""
+import logging
+
+
+class LogOperations:
+    """:class:`~.LogOperations` logs information about the execution of operations to a log file.
+
+    This hooks provides information, optionally, on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+
+    The default formating for the log provides the time, job id, log level, and log message.
+
+    .. note::
+
+        All logging is performed at the INFO level. To ensure outputs are captured in log files,
+        use the `--debug` flag when running or submitting jobs, or specify
+        `submit_options=--debug` in your directives (example shown below).
+
+
+    Examples
+    --------
+    The following example will install :class:`~.LogOperations` at the operation level.
+    Where the log will be stored in a file name `foo.log` in the job workspace.
```

Comment:
Echoing @bdice 's comment, the term "job workspace" might need to be adjusted.

Reply:
https://docs.signac.io/en/latest/glossary.html#term-job-directory